### PR TITLE
M3-5196: React Query for Volumes - Volumes Landing

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- The goal with **part 1** is to get the Volumes Landing and all associated actions such as attach, delete, etc working

## How to test 🧪

- Test **only** Volumes Landing features at `http://localhost:3000/volumes`
  - Check all actions/mutations possible in the Volumes Landing Action Menus
  - Check that Landing table updates accordingly following actions such as attach, delete, clone, etc...

### CODE REVIEW

File: packages/manager/src/queries/base.ts [hunk 1]
```diff
@@ -200,3 +200,35 @@ export const itemInListDeletionHandler = <
...
+
+      oldData.data[toUpdateIndex] = {
+        ...oldData.data[toUpdateIndex],
+        ...newData,
+      };
+
+      return oldData;
+    }
+  );
+};
```
Comment (CodyFinn-Akamai) [inline packages/manager/src/queries/base.ts:234]: Nice job abstracting this out using generics!

**Review by CodyFinn-Akamai — APPROVED** [review]

**Comment by bnussman-akamai** [comment]
@jdamore-linode Can you check e2es on this?

File: packages/manager/src/queries/volumes.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    }
+  );
+
+export const useDetachVolumeMutation = () =>
+  useMutation<{}, APIError[], { id: number }>(({ id }) => detachVolume(id));
+
+export const volumeEventsHandler = (event: Event) => {
+  const { action, status, entity } = event;
+
+  switch (action) {
```
Comment (bnussman-akamai) [inline packages/manager/src/queries/volumes.ts:91]: I will make these `switch` statements this more readable.

**Review by jdamore-linode — APPROVED** [review]
Nice! E2E tests all pass, and I messed around with the Volumes landing page and confirmed that each operation works as expected

**Pull request merged**